pyvrp/cpp/ProblemData.cpp

```diff
@@ -16,26 +16,14 @@ size_t ProblemData::numVehicles() const { return numVehicles_; }
 
 size_t ProblemData::vehicleCapacity() const { return vehicleCapacity_; }
 
-ProblemData::ProblemData(std::vector<std::pair<int, int>> const &coords,
-                         std::vector<int> const &demands,
+ProblemData::ProblemData(std::vector<Client> const &clients,
                          size_t numVehicles,
                          size_t vehicleCap,
-                         std::vector<std::pair<int, int>> const &timeWindows,
-                         std::vector<int> const &servDurs,
                          std::vector<std::vector<int>> const &distMat)
     : dist_(distMat),
-      clients_(coords.size()),
-      numClients_(static_cast<int>(coords.size()) - 1),
+      clients_(clients),
+      numClients_(clients.size() - 1),
```

Comment:
I only notice this now that we're changing it, but this could potentially be a really large number if size == 0. Can we do something like
```cpp
numClients_(std::max(clients.size(), 1) - 1),
```
to ensure this works fine when creating an empty instance?